march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -430,23 +439,20 @@ void MarchHardwareInterface::updateIMotionCubeState()
   imc_state_pub_->unlockAndPublish();
 }
 
-void MarchHardwareInterface::iMotionCubeStateCheck(size_t joint_index)
+bool MarchHardwareInterface::iMotionCubeStateCheck(size_t joint_index)
 {
   march::Joint& joint = march_robot_->getJoint(joint_index);
   march::IMotionCubeState imc_state = joint.getIMotionCubeState();
   if (imc_state.state == march::IMCState::FAULT)
   {
-    this->march_robot_->stopEtherCAT();
-    std::ostringstream error_stream;
-    error_stream << "IMotionCube of joint " << joint.getName() << " is in fault state " << imc_state.state.getString()
-                 << std::endl;
-    error_stream << "Detailed Error: " << imc_state.detailedErrorDescription << "(" << imc_state.detailedError << ")"
-                 << std::endl;
-    error_stream << "Motion Error: " << imc_state.motionErrorDescription << "(" << imc_state.motionError << ")"
-                 << std::endl;
-
-    throw std::runtime_error(error_stream.str());
+    ROS_ERROR("IMotionCube of joint %d is in fault state %s \n", joint.getName().c_str(),
+              imc_state.state.getString().c_str());
```

Comment:
Found a tiny teensy oopsie. `%d` will format for integers, however, the first parameter is a string.
```suggestion
    ROS_ERROR("IMotionCube of joint %s is in fault state %s \n", joint.getName().c_str(),
              imc_state.state.getString().c_str());
```